Assets/_BForBoss/_Core/Scripts/Managers/WaveManager.cs

```diff
@@ -55,12 +55,12 @@ private void OnEnemySpawned()
             }
         }
 
-        private void OnEnemyKilled(int killCount)
+        private void OnEnemyKilled(int numberOfRemainingEnemies)
         {
             //Added the second clause just in case no. spawned > max allowed
-            if (killCount >= _waveModel.MaxEnemyCount && killCount == _spawnCount)
+            if (numberOfRemainingEnemies <= 0 && _waveModel.KillCount == _spawnCount)
```

Comment:
`_waveModel.KillCount == _spawnCount` is this still really needed?

Reply:
Technically no, but this is to really ensure that the spawn count and KillCount are in total balance (as a sanity check).  Up to you though, if you want me to remove it

Reply:
Seeing how it's always going to go down, i don't really see a point of having it anymore.